resources-domain/lambdas/s3-importer/src/index.ts

```diff
@@ -15,7 +15,66 @@
  */
 
 import type { S3Event } from 'aws-lambda';
+import { parse } from 'csv-parse';
+import { deleteS3Object, getS3Object, putS3Object } from '@tech-matters/s3-client';
+import { expandCsvLine, transformUschResourceToApiResource } from './uschMappings';
+import {
+  waitForEmptyQueue,
+  publishToImportConsumer,
+  ResourceMessage,
+} from '@tech-matters/resources-import-producer';
+import getConfig from './config';
+import { newSqsClient } from '@tech-matters/sqs-client';
+
+export const COMPLETED_KEY = 'completed-s3-imports';
 
 export const handler = async (event: S3Event): Promise<void> => {
   console.debug('Triggered by event:', JSON.stringify(event));
+  const { largeMessagesS3Bucket, importResourcesSqsQueueUrl, accountSid } =
+    await getConfig();
+
+  const sqs = newSqsClient({
+    largeMessageS3BaseLocation: {
+      bucket: largeMessagesS3Bucket,
+      key: 'sqsLargeMessageContent',
+    },
+  });
+  const configuredPublish = publishToImportConsumer(sqs, importResourcesSqsQueueUrl);
+
+  await waitForEmptyQueue(importResourcesSqsQueueUrl);
```

Comment:
Why does this awaits the queue to be empty? Can this lead to an execution that reaches execution timeout without doing any actual work?